### PR TITLE
Enabled custom networking for AWS instances.

### PR DESCRIPTION
**Pull request by narmaku**
Other changes include:
- Fixed html report display glitch reported in CLOUDX-316.
- Extended instance metadata to include private IP upon deployment.
- Minor refactoring to make the code more adaptable.

### CODE REVIEW

**Review by ConorC117 — APPROVED** [review]
Everything looks good to me!

File: cloud/opentofu/opentofu_controller.py [hunk 1]
```diff
@@ -92,11 +92,24 @@ def get_instances_aws(self, resources):
                 'instance_id': resource['values']['id'],
                 'public_ip': resource['values']['public_ip'],
                 'public_dns': resource['values']['public_dns'],
+                'private_ip': resource['values']['private_ip'],
                 'availability_zone': resource['values']['availability_zone'],
                 'ami': ami_name,
+                'image': ami_name,
```
[the remark this answers is not in the document]
Comment (narmaku) [reply cloud/opentofu/opentofu_controller.py:98]: To make it more generic and flexible. We need this to render the "image" column correctly in the HTML report.

File: cloud/opentofu/opentofu_controller.py [hunk 2]
```diff
@@ -145,7 +159,8 @@ def get_instances_gcloud(self, resources):
                 'name': resource['name'],
                 'instance_id': resource['values']['id'],
                 'public_ip': public_ip,
-                'public_dns': public_ip,
+                'public_dns': public_ip,  # TODO: Support also public dns as we do for the other Clouds
+                'address': public_ip,  # TODO: Support also private IP addresses as we do for AWS
```
[the remark this answers is not in the document]
Comment (narmaku) [reply cloud/opentofu/opentofu_controller.py:163]: In the deployment we do for GCP, we only deploy with public IP address. So by default I added the public IP to the address, and the private IP is not added (yet). If we start working actively on testing GCP images, we have to revisit this. Therefore I left the TODO items there.

**Review by FKolwa — COMMENTED** [review]
Looks good to me too.
I like that we're finally getting our own network done.

**Pull request merged**